### DISCUSSION TITLE
Creating non-circle based curve lines

### DISCUSSION THREAD

**Discussion by Andrej730** [IfcOpenShell/IfcOpenShell#2663] [Q&A]
To create curve lines in IFC I mostly use IfcArcIndex - it works and it's really good for creating fillets. But sometimes I need curve lines that are not based on a single radius.

For example if I try to create IfcArcIndex between points A and B I get the circular curve from the picture.
When my goal is to get more ellipse shaped curve like the red line. How can I achieve that line in ifc? Perhaps there is some example either in python/c++/ifc?
It also would be great if it won't be just ellipse based and I would also be able to modify it a bit and get something like the yellow line for example (like if I'd be able to specify exact points to base curve on).
It seems like there is some simple solution I'm missing😅

![image](https://user-images.githubusercontent.com/9417531/209471530-46b915ac-20ba-4a96-bfbd-138c42b37601.png)

PS I've tried to create `IfcEllipse` and then trim it with `IfcTrimmedCurve` - it works but I'm not sure 1) how can I rotate that shape if needed; 2) how do I modify it to make it a bit different shape then ellipse. And it's seems there's gotta be simpler way.

**Comment by Moult** [comment]
What's the ellipse for? And is it actually an ellipse or is it just a circle with a greater radius?

And yes what you've done with IfcEllipse with IfcTrimmedCurve sounds correct.

**Reply by Andrej730** [reply]
It's an actual ellipse, it would be more convenient to make shapes like this instead of circle based arc.
![image](https://user-images.githubusercontent.com/9417531/209606010-cd91aa45-71d7-48aa-8c24-dd6d3e1af9ef.png)
